### PR TITLE
Update org logo API

### LINKED ISSUE #66: update profile_pic api
**Is your feature request related to a problem? Please describe.**
As a admin of the org I should be able to update the profile_pic/logo of the org.

**Describe the solution you'd like**
1. Create api [put] /api/org/(org-id)/profile_pic/ which takes in new pic which replaces the old one.
1. Make sure to delete the old pic.
1. Only admin should able to perform this task.
1. Add appropriate tests.
1. Add swagger docs for the same.

### PR DESCRIPTION
**Pull request by akhil-maker**
### Description
I have added a PUT API to update the profile-pic of an organisation. Only test cases are not provided for this pr.

**Type of Change:** (Delete irrelevant options)
- Code
- Testing
- User Interface

**Code/Quality Assurance Only** (Delete irrelevant options)

- Bug fix (non-breaking change which fixes an issue)
- This change requires a documentation update (software upgrade on readme file)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
Tested locally with the help of postman 

When all details provided by an authenticated and authorized user who is the admin of the organisation.
IMG_20200604_093122

It will provide following response when authentication details are not provided.
IMG_20200604_092851

When the organisation is deleted or not present, it will show the following response
IMG_20200604_092729

### Checklist:
- [X] My PR follows the style guidelines of this project
- [X] I have performed a self-review of my own code or materials
- [X] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged
- [X] Update Swagger documentation and the exported file at /docs folder
- [ ] Update requirements.txt

**Code/Quality Assurance**
- [X] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Comment by sahil9001** [comment]
Few changes required,
1. Add test cases (without them we can't merge)
2. Make a new branch for the issue, we don't merge `dev` branch of your fork.
3. Resolve conflicts.
4. Rebase or either pull the upstream `dev` to fetch latest changes.

File: org/urls.py [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 urlpatterns = [
     path('', OrgView.as_view()),
+    path('<int:org_id>/', put_pic),
```
Comment (sahil9001) [inline org/urls.py:7]: This should be `<int:org_id>/profile_pic/` I guess.

File: org/serializers.py [hunk 2]
```diff
@@ -76,3 +76,7 @@ def save(self):
 
         return [org,admin_group]
 
+class picserializer(serializers.ModelSerializer):
```
Comment (sahil9001) [inline org/serializers.py:79]: Naming of serializers should be proper.

File: org/responses.py [hunk 3]
```diff
@@ -40,3 +40,42 @@
...
+    "message": "Organization details updated successfully"
+}
+
+admin_access_403 = {
+    "message": "You are not an admin of this organization"
+}
+
+empty_fields_500 = {
+    "message": "Can't update! one or more fields are empty!"
+}
```
Comment (sahil9001) [inline org/responses.py]: I see some repeated responses here, please remove them.

**Review by sahil9001 — CHANGES_REQUESTED** [review]

**Comment by naveennvrgup** [comment]
> 1. Add test cases (without them we can't merge)
1. Please add a folder PROJECT_ROOT/assets/test/
1. Add a pic over there
1. and try to send that pic while making the testing
1. testing is crucial as @sahil9001 told
> 2. Make a new branch for the issue, we don't merge `dev` branch of your fork.
> 3. Resolve conflicts.
> 4. Rebase or either pull the upstream `dev` to fetch latest changes.

(2-4)absolutely. I recommend `pull the upstream dev to feature branch`  for beginners.

File: org/views.py [hunk 4]
```diff
@@ -135,5 +135,40 @@ def GetGroup(request,org_id):
...
-    
\ No newline at end of file
+    try:
+        member = Member.objects.get(user=user, org=org)
+    except Member.DoesNotExist:
+        return Response(responses.admin_access_403, status.HTTP_403_FORBIDDEN)
+
+    isadmin = member.group.perm_obj.permissions_to_integer()
+    # Checking if the isadmin is odd or even, if odd then the IS_ADMIN permission is enabled for the user
+    if isadmin % 2 == 1:
```
Comment (anshsrtv) [inline org/views.py:165]: @akhil-maker Please explain the condition.. I didn't get your way of checking whether the user is an Admin

File: org/views.py [hunk 5]
```diff
@@ -135,5 +135,40 @@ def GetGroup(request,org_id):
...
+        if request.method == "PUT":
```
Comment (anshsrtv) [inline org/views.py:166]: I don't think there's any need for checking whether the method is `PUT`. It's already taken care of by the `@api_view(['PUT'])` .

File: org/views.py [hunk 6]
```diff
@@ -135,5 +135,40 @@ def GetGroup(request,org_id):
...
+            org_data = FormParser().parse(request)
+            pic_serializer = picserializer(org, data=org_data)
+            if pic_serializer.is_valid():
+                pic_serializer.save()
+                return Response(responses.update_org_200, status.HTTP_200_OK)
+            return Response(pic_serializer.errors, status.HTTP_400_BAD_REQUEST)
+        else:
```
Comment (anshsrtv) [inline org/views.py:173]: A mistake, check the indentation.

File: org/serializers.py (hunk 2 above)
Comment (anshsrtv) [reply org/serializers.py:79]: Try naming variables the way we usually name them.. looks good that way :wink:

**Review by anshsrtv — CHANGES_REQUESTED** [review]
Wasn't hoping for this PR, so Kudos :tada:, you made the PR . But surely a lot of work left...

1. The responses aren't proper, the API gives `"Organization details updated successfully"` everytime an admin makes a request.
2. Check if the API works properly using the admin panel.
4. Hope the testcases arrive soon :v:

File: org/serializers.py (hunk 2 above)
Comment (akhil-maker) [reply org/serializers.py:79]: Ok @sahil9001 @anshsrtv

File: org/views.py (hunk 6 above)
Comment (akhil-maker) [reply org/views.py:173]: I will correct it @anshsrtv

**Pull request closed without merging**